packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -3451,6 +3539,86 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     return TextSelection(baseOffset: line.start, extentOffset: line.end);
   }
 
+  // Placeholder dimensions representing the sizes of child inline widgets.
+  //
+  // These need to be cached because the text painter's placeholder dimensions
+  // will be overwritten during intrinsic width/height calculations and must be
+  // restored to the original values before final layout and painting.
+  List<PlaceholderDimensions>? _placeholderDimensions;
+
+  // Layout the child inline widgets. We then pass the dimensions of the
```

Comment:
A substantial amount of (somewhat complex) logic here seems to be shared with RenderParagraph for similar purposes. I wonder if we can find a way to share it between the implementations.

Reply:
(true for other methods in this file as well)

Reply:
Some of the work in https://github.com/flutter/flutter/issues/79420 may make it easier to support a lot of this logic in a common place by removing many of the editing methods, but as of now, reuse of code would require a fairly complex redesign/refactor.